data-prepper-core/src/main/java/com/amazon/dataprepper/parser/PipelineParser.java

```diff
@@ -8,8 +8,10 @@
 import com.amazon.dataprepper.model.annotations.SingleThread;
 import com.amazon.dataprepper.model.buffer.Buffer;
 import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.plugin.NoPluginFoundException;
 import com.amazon.dataprepper.model.plugin.PluginFactory;
 import com.amazon.dataprepper.model.prepper.Prepper;
```

Comment:
Do we need Prepper import ?

Reply:
Looks like we need this as fallback when processor type plugin is not found.

Reply:
Yes. The current implementation of the @DataPrepperPlugin annotation and how plugin instances are created requires an exact class match on the pluginType property. Until all Prepper plugins have migrated to the Processor class plugin type the newProcessor will need to be able to fallback to casting a new Prepper instance to a Processor.

Alternative options would be:
1. Update all plugin annotations within this PR
2. Rework how plugin instances are loaded
3. Hard code a conversion of prepper to processor within the DefaultPluginFactory

Fallback process: https://github.com/sbayer55/data-prepper/blob/rename-prepper-to-processor/data-prepper-core/src/main/java/com/amazon/dataprepper/parser/PipelineParser.java#L127

Plugin factory class matching: https://github.com/sbayer55/data-prepper/blob/rename-prepper-to-processor/data-prepper-core/src/main/java/com/amazon/dataprepper/plugin/DefaultPluginFactory.java#L104